ember-core/shared/src/main/scala/org/http4s/ember/core/ChunkedEncoding.scala

```diff
@@ -140,20 +142,37 @@ private[ember] object ChunkedEncoding {
     }
 
   private val lastChunk: Chunk[Byte] =
-    Chunk.byteVector((ByteVector('0') ++ crlf ++ crlf).compact)
+    Chunk.byteVector((ByteVector('0') ++ crlf).compact)
 
   /** Encodes chunk of bytes to http chunked encoding.
     */
-  def encode[F[_]]: Pipe[F, Byte, Byte] = {
+  def encode[F[_]: Functor](trailers: F[Headers]): Pipe[F, Byte, Byte] = {
     def encodeChunk(bv: ByteVector): Chunk[Byte] =
       if (bv.isEmpty) Chunk.empty
       else
         Chunk.byteVector(
           ByteVector.view(bv.size.toHexString.toUpperCase.getBytes) ++ crlf ++ bv ++ crlf
         )
-    _.mapChunks { ch =>
-      encodeChunk(ch.toByteVector)
-    } ++ Stream.chunk(lastChunk)
+    _.mapChunks(ch => encodeChunk(ch.toByteVector)) ++
+      Stream.chunk(lastChunk) ++
+      Stream.evalUnChunk(trailers.map(encodeTrailers(_))) ++
+      Stream.chunk(Chunk.byteVector(crlf))
```

Comment:
We can make some additional optimizations here. Just waiting to see whether this PR or https://github.com/http4s/http4s/pull/6753 will merge first.